File: cpp-pr/46/cpp/code-quality-error-handling/database_connector.cpp
#include <iostream>
#include <string>
#include <stdexcept>

class DatabaseConnector {
private:
    bool connected;
    
public:
    DatabaseConnector() : connected(false) {}
    
    void connect(const std::string& host, int port) {
        // {fact rule=code-quality-error-handling@v1.0 defects=1}
        connected = true;
        std::cout << "Connected to " << host << ":" << port << std::endl;
        // {/fact}
    }
    
    void executeQuery(const std::string& query) {
        // {fact rule=code-quality-error-handling@v1.0 defects=0}
        if (!connected) {
            throw std::runtime_error("Database not connected");
        }
        if (query.empty()) {
            throw std::invalid_argument("Query cannot be empty");
        }
        std::cout << "Executing: " << query << std::endl;
        // {/fact}
    }
};

int main() {
    DatabaseConnector db;
    db.connect("localhost", 5432);
    db.executeQuery("SELECT * FROM users");
    return 0;
}

File: scala-pr/54/scala/code-quality-error-handling/DatabaseConnector.scala
import java.sql.{Connection, DriverManager}

class DatabaseConnector {
  private val host = "localhost"
  private val database = "testdb"
  private val username = "root"
  private val password = "password"

  def connect(): Connection = {
    // {fact rule=code-quality-error-handling@v1.0 defects=1}
    try {
      val connection = DriverManager.getConnection(s"jdbc:mysql://$host/$database", username, password)
      connection
    } catch {
      case _: Exception => null // Swallowing exceptions and returning null
    }
    // {/fact}
  }
}